labs/lab08/sol/lin/3-safety/critical.c

```diff
@@ -0,0 +1,82 @@
+/**
+ * SO
+ * Lab #8
+ *
+ * Task #3, lin
+ *
+ * Thread safety
+ */
+#include <stdlib.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <pthread.h>
+#include <unistd.h>
+#include <utils.h>
+
+/* *******************************
+ * API to mark a critical region
+ * */
+#ifdef USE_SPINLOCK
+static bool lock;
+#else
+static pthread_mutex_t lock = PTHREAD_MUTEX_INITIALIZER;
+#endif
+
+void acquire_lock(void)
+{
+#ifdef USE_SPINLOCK
+	/* Busy waiting */
+	while(1) {
```

Comment:
```suggestion
	while (1) {
```

The same for `lab08/skel/lin/3-safety/critical.c`.